src/base/bittorrent/trackerentry.h

```diff
@@ -47,26 +50,21 @@ namespace BitTorrent
             NotWorking = 4
         };
 
-        TrackerEntry() = default;
-        TrackerEntry(const QString &url);
-        TrackerEntry(const lt::announce_entry &nativeEntry);
-        TrackerEntry(const TrackerEntry &other) = default;
-        TrackerEntry &operator=(const TrackerEntry &other) = default;
-
-        QString url() const;
-        Status status() const;
-
-        int tier() const;
-        void setTier(int value);
-
-        int numSeeds() const;
-        int numLeeches() const;
-        int numDownloaded() const;
+        struct Endpoint
+        {
+            int protocolVersion = 1;
+            Status status = NotContacted;
+            int numSeeds = -1;
+            int numLeeches = -1;
+            int numDownloaded = -1;
+        };
 
-        const lt::announce_entry &nativeEntry() const;
+        QVector<Endpoint> endpoints;
```

Comment:
I would layout the struct variables as:
```c++
struct TrackerEntry
{
    QString url;
    int tier = 0;

    Status status = NotContacted;
    int numSeeds = -1;
    int numLeeches = -1;
    int numDownloaded = -1;
    QVector<Endpoint> endpoints;
}
```

Reply:
I just put deprecated entries in the end. Maybe I can get rid of them completely even in this PR, without postponing it for later (see about summary data calculation in PR description).